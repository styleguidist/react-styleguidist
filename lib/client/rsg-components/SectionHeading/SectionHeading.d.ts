import React from 'react';
interface SectionHeadingProps {
    children?: React.ReactNode;
    id: string;
    slotName: string;
    slotProps: Record<string, unknown>;
    depth: number;
    href?: string;
    deprecated?: boolean;
    pagePerSection?: boolean;
}
declare const SectionHeading: React.FunctionComponent<SectionHeadingProps>;
export default SectionHeading;
