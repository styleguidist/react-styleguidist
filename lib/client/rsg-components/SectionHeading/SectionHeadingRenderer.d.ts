import React from 'react';
import { JssInjectedProps } from 'rsg-components/Styled';
interface SectionHeadingRendererProps extends JssInjectedProps {
    children?: React.ReactNode;
    toolbar?: React.ReactNode;
    id: string;
    href?: string;
    depth: number;
    deprecated?: boolean;
}
declare const _default: React.ComponentType<Pick<SectionHeadingRendererProps, "toolbar" | "children" | "id" | "href" | "deprecated" | "depth">>;
export default _default;
