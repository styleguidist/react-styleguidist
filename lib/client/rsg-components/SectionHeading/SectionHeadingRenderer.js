import React from 'react';
import PropTypes from 'prop-types';
import cx from 'clsx';
import Heading from 'rsg-components/Heading';
import Styled from 'rsg-components/Styled';

var styles = function styles(_ref) {
  var color = _ref.color,
      space = _ref.space;
  return {
    wrapper: {
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
      marginBottom: space[1]
    },
    toolbar: {
      marginLeft: 'auto'
    },
    sectionName: {
      '&:hover, &:active': {
        isolate: false,
        textDecoration: 'underline',
        cursor: 'pointer'
      }
    },
    isDeprecated: {
      color: color.light,
      '&, &:hover': {
        textDecoration: 'line-through'
      }
    }
  };
};

var SectionHeadingRenderer = function SectionHeadingRenderer(_ref2) {
  var _cx;

  var classes = _ref2.classes,
      children = _ref2.children,
      toolbar = _ref2.toolbar,
      id = _ref2.id,
      href = _ref2.href,
      depth = _ref2.depth,
      deprecated = _ref2.deprecated;
  var headingLevel = Math.min(6, depth);
  var sectionNameClasses = cx(classes.sectionName, (_cx = {}, _cx[classes.isDeprecated] = deprecated, _cx));
  return /*#__PURE__*/React.createElement("div", {
    className: classes.wrapper
  }, /*#__PURE__*/React.createElement(Heading, {
    level: headingLevel,
    id: id
  }, /*#__PURE__*/React.createElement("a", {
    href: href,
    className: sectionNameClasses
  }, children)), /*#__PURE__*/React.createElement("div", {
    className: classes.toolbar
  }, toolbar));
};

SectionHeadingRenderer.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
  children: PropTypes.node,
  toolbar: PropTypes.node,
  id: PropTypes.string.isRequired,
  href: PropTypes.string,
  depth: PropTypes.number.isRequired,
  deprecated: PropTypes.bool
};
export default Styled(styles)(SectionHeadingRenderer);