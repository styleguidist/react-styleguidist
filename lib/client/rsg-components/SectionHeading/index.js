export { default } from 'rsg-components/SectionHeading/SectionHeading';