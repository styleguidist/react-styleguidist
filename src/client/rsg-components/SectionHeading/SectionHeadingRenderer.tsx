import React from 'react';
import PropTypes from 'prop-types';
import cx from 'clsx';
import Heading from 'rsg-components/Heading';
import Styled, { JssInjectedProps } from 'rsg-components/Styled';
import * as Rsg from '../../../typings';

const styles = ({ color, space }: Rsg.Theme) => ({
	wrapper: {
		display: 'flex',
		flexDirection: 'row',
		alignItems: 'center',
		marginBottom: space[1],
	},
	toolbar: {
		marginLeft: 'auto',
	},
	sectionName: {
		'&:hover, &:active': {
			isolate: false,
			textDecoration: 'underline',
			cursor: 'pointer',
		},
	},
	isDeprecated: {
		color: color.light,
		'&, &:hover': {
			textDecoration: 'line-through',
		},
	},
});

interface SectionHeadingRendererProps extends JssInjectedProps {
	children?: React.ReactNode;
	toolbar?: React.ReactNode;
	id: string;
	href?: string;
	depth: number;
	deprecated?: boolean;
}

const SectionHeadingRenderer: React.FunctionComponent<SectionHeadingRendererProps> = ({
	classes,
	children,
	toolbar,
	id,
	href,
	depth,
	deprecated,
}) => {
	const headingLevel = Math.min(6, depth);
	const sectionNameClasses = cx(classes.sectionName, {
		[classes.isDeprecated]: deprecated,
	});

	return (
		<div className={classes.wrapper}>
			<Heading level={headingLevel} id={id}>
				<a href={href} className={sectionNameClasses}>
					{children}
				</a>
			</Heading>
			<div className={classes.toolbar}>{toolbar}</div>
		</div>
	);
};

SectionHeadingRenderer.propTypes = {
	classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
	children: PropTypes.any,
	toolbar: PropTypes.any,
	id: PropTypes.string.isRequired,
	href: PropTypes.string,
	depth: PropTypes.number.isRequired,
	deprecated: PropTypes.bool,
};

export default Styled<SectionHeadingRendererProps>(styles)(SectionHeadingRenderer);
