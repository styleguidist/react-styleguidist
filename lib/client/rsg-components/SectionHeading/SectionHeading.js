import "core-js/modules/es.array.index-of";
import "core-js/modules/es.object.assign";
import "core-js/modules/es.object.keys";

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import PropTypes from 'prop-types';
import Slot from 'rsg-components/Slot';
import SectionHeadingRenderer from 'rsg-components/SectionHeading/SectionHeadingRenderer';

var SectionHeading = function SectionHeading(_ref) {
  var slotName = _ref.slotName,
      slotProps = _ref.slotProps,
      children = _ref.children,
      id = _ref.id,
      href = _ref.href,
      rest = _objectWithoutPropertiesLoose(_ref, ["slotName", "slotProps", "children", "id", "href"]);

  return /*#__PURE__*/React.createElement(SectionHeadingRenderer, _extends({
    toolbar: /*#__PURE__*/React.createElement(Slot, {
      name: slotName,
      props: slotProps
    }),
    id: id,
    href: href
  }, rest), children);
};

SectionHeading.propTypes = {
  children: PropTypes.node,
  id: PropTypes.string.isRequired,
  slotName: PropTypes.string.isRequired,
  slotProps: PropTypes.any.isRequired,
  depth: PropTypes.number.isRequired,
  deprecated: PropTypes.bool,
  pagePerSection: PropTypes.bool
};
export default SectionHeading;