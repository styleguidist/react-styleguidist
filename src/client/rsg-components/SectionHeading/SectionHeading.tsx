import React from 'react';
import PropTypes from 'prop-types';
import Slot from 'rsg-components/Slot';
import SectionHeadingRenderer from 'rsg-components/SectionHeading/SectionHeadingRenderer';

interface SectionHeadingProps {
	children?: React.ReactNode;
	id: string;
	slotName: string;
	slotProps: Record<string, unknown>;
	depth: number;
	href?: string;
	deprecated?: boolean;
	pagePerSection?: boolean;
}

const SectionHeading: React.FunctionComponent<SectionHeadingProps> = ({
	slotName,
	slotProps,
	children,
	id,
	href,
	...rest
}) => {
	return (
		<SectionHeadingRenderer
			toolbar={<Slot name={slotName} props={slotProps} />}
			id={id}
			href={href}
			{...rest}
		>
			{children}
		</SectionHeadingRenderer>
	);
};

SectionHeading.propTypes = {
	children: PropTypes.any,
	id: PropTypes.string.isRequired,
	slotName: PropTypes.string.isRequired,
	slotProps: PropTypes.any.isRequired,
	depth: PropTypes.number.isRequired,
	deprecated: PropTypes.bool,
	pagePerSection: PropTypes.bool,
};

export default SectionHeading;
