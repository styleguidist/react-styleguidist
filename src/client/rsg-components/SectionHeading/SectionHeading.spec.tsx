import React from 'react';
import renderer from 'react-test-renderer';
import { createRenderer } from 'react-test-renderer/shallow';
import SectionHeading from './index';
import SectionHeadingRenderer from './SectionHeadingRenderer';

describe('SectionHeading', () => {
	const FakeToolbar = () => <div>Fake toolbar</div>;

	test('should forward slot properties to the toolbar', () => {
		const renderer = createRenderer();
		renderer.render(
			<SectionHeading
				id="section"
				slotName="slot"
				href="/#section"
				slotProps={{ foo: 1, bar: 'baz' }}
				depth={2}
			>
				A Section
			</SectionHeading>
		);

		expect(renderer.getRenderOutput()).toMatchSnapshot();
	});

	test('render a section heading', () => {
		const actual = renderer.create(
			<SectionHeadingRenderer id="section" href="/section" depth={2} toolbar={<FakeToolbar />}>
				A Section
			</SectionHeadingRenderer>
		);

		expect(actual.toJSON()).toMatchSnapshot();
	});

	test('render a deprecated section heading', () => {
		const actual = renderer.create(
			<SectionHeadingRenderer
				id="section"
				href="/section"
				depth={2}
				toolbar={<FakeToolbar />}
				deprecated
			>
				A Section
			</SectionHeadingRenderer>
		);

		expect(actual.toJSON()).toMatchSnapshot();
	});

	test('prevent the heading level from exceeding the maximum allowed by the Heading component', () => {
		const actual = renderer.create(
			<SectionHeadingRenderer id="section" href="/section" depth={7} toolbar={<FakeToolbar />}>
				A Section
			</SectionHeadingRenderer>
		);

		expect(actual.toJSON()).toMatchSnapshot();
	});
});
