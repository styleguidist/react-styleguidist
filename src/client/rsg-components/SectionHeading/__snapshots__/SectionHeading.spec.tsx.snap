// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeading prevent the heading level from exceeding the maximum allowed by the Heading component 1`] = `
<div
  className="rsg--wrapper-0"
>
  <h6
    className="rsg--heading-5 rsg--heading6-11"
    id="section"
  >
    <a
      className="rsg--sectionName-2"
      href="/section"
    >
      A Section
    </a>
  </h6>
  <div
    className="rsg--toolbar-1"
  >
    <div>
      Fake toolbar
    </div>
  </div>
</div>
`;

exports[`SectionHeading render a deprecated section heading 1`] = `
<div
  className="rsg--wrapper-0"
>
  <h2
    className="rsg--heading-5 rsg--heading2-7"
    id="section"
  >
    <a
      className="rsg--sectionName-2 rsg--isDeprecated-3"
      href="/section"
    >
      A Section
    </a>
  </h2>
  <div
    className="rsg--toolbar-1"
  >
    <div>
      Fake toolbar
    </div>
  </div>
</div>
`;

exports[`SectionHeading render a section heading 1`] = `
<div
  className="rsg--wrapper-0"
>
  <h2
    className="rsg--heading-5 rsg--heading2-7"
    id="section"
  >
    <a
      className="rsg--sectionName-2"
      href="/section"
    >
      A Section
    </a>
  </h2>
  <div
    className="rsg--toolbar-1"
  >
    <div>
      Fake toolbar
    </div>
  </div>
</div>
`;

exports[`SectionHeading should forward slot properties to the toolbar 1`] = `
<Styled(SectionHeading)
  depth={2}
  href="/#section"
  id="section"
  toolbar={
    <Slot
      name="slot"
      props={
        Object {
          "bar": "baz",
          "foo": 1,
        }
      }
    />
  }
>
  A Section
</Styled(SectionHeading)>
`;
